astropy/convolution/core.py

```diff
@@ -88,42 +85,66 @@ def center(self):
         """
         return [axes_size // 2 for axes_size in self._array.shape]
 
+    def calculate_normalization(self, mode='integral'):
+        """
+        Calculate the kernel normalization factor.
+
+        Parameters
+        ----------
+        mode : {'integral', 'peak'}
+            One of the following modes:
+                * 'integral' (default)
+                    Kernel is normalized such that its integral = 1.
+                * 'peak'
+                    Kernel is normalized such that its peak = 1.
+        """
+
+        if mode == 'integral':
+            if self._array.sum() == 0:
+                self._normalization = np.inf
+            else:
+                self._normalization = 1. / self._array.sum()
+        elif mode == 'peak':
+            self._normalization = 1. / self._array.max()
+        else:
+            raise ValueError("invalid mode, must be 'integral' or 'peak'")
+        return self._normalization
+
     @property
     def normalization(self):
         """
-        Kernel normalization factor
+        The kernel normalization factor.
         """
         return self._normalization
 
     def normalize(self, mode='integral'):
         """
-        Force normalization of filter kernel.
+        Normalize the filter kernel.
 
         Parameters
         ----------
         mode : {'integral', 'peak'}
             One of the following modes:
                 * 'integral' (default)
-                    Kernel normalized such that its integral = 1.
+                    Kernel is normalized such that its integral = 1.
                 * 'peak'
-                    Kernel normalized such that its peak = 1.
+                    Kernel is normalized such that its peak = 1.
         """
-        # There are kernel that sum to zero and
-        # the user should be warned in this case
+
+        np.multiply(self._array, self.calculate_normalization(mode=mode),
+                    self.array)
+
+        # Warn the user for kernels that sum to zero
         if np.isinf(self._normalization):
             warnings.warn('Kernel cannot be normalized because the '
                           'normalization factor is infinite.',
                           AstropyUserWarning)
             return
+
         if np.abs(self._normalization) > MAX_NORMALIZATION:
-            warnings.warn("Normalization factor of kernel is "
-                          "exceptionally large > {0}.".format(MAX_NORMALIZATION),
+            warnings.warn('The kernel normalization factor is exceptionally '
+                          'large, > {0}.'.format(MAX_NORMALIZATION),
                           AstropyUserWarning)
-        if mode == 'integral':
-            self._array *= self._normalization
```

Comment:
I think adding `self._normalization = 1` or `self._normalization = 1. / self._array.sum()` would fix the bug